From 1b0912d161c301e5a806fb44b9220bd486affe93 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Tue, 13 May 2014 12:04:34 -0700
Subject: [PATCH 473/518] MIPS: Don't used EHB in TLB handlers when not needed.

Source: Cavium Networks, Inc.
MR: 00000
Type: Integration
Disposition: Merged from Octeon Tree
ChangeID: ca65c65537d8bdd265695d3478547d1cedbc785e
Description:

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
[Original patch taken from Cavium SDK 3.1.1 525]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/mm/tlbex.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/mips/mm/tlbex.c b/arch/mips/mm/tlbex.c
index 34494ff..f88c8f5 100644
--- a/arch/mips/mm/tlbex.c
+++ b/arch/mips/mm/tlbex.c
@@ -526,7 +526,8 @@ static void __cpuinit build_tlb_write_entry(u32 **p, struct uasm_label **l,
 			break;
 
 		default:
-			uasm_i_ehb(p);
+			if (cpu_has_mips_r2_exec_hazard)
+				uasm_i_ehb(p);
 			break;
 		}
 		tlbw(p);
-- 
1.9.1

